resources/backup-vaults-access-policies.go

```diff
@@ -0,0 +1,120 @@
+	register("AWSBackupVaultAccessPolicy", ListBackupVaultAccessPolicies)
+}
+
+func ListBackupVaultAccessPolicies(sess *session.Session) ([]Resource, error) {
+	svc := backup.New(sess)
+	maxVaultsLen := int64(100)
+	params := &backup.ListBackupVaultsInput{
+		MaxResults: &maxVaultsLen, // aws default limit on number of backup vaults per account
+	}
+	resp, err := svc.ListBackupVaults(params)
+	if err != nil {
+		return nil, err
+	}
+
+	// Iterate over backup vaults and add vault policies that exist.
+	resources := make([]Resource, 0)
+	for _, out := range resp.BackupVaultList {
+		// Check if the Backup Vault has an Access Policy set
+		resp, err := svc.GetBackupVaultAccessPolicy(&backup.GetBackupVaultAccessPolicyInput{
+			BackupVaultName: out.BackupVaultName,
+		})
+
+		// Non-existent Access Policies can come from ResourceNotFoundException or
+		// being nil.
+		if err != nil {
+			switch err.(type) {
+			case *backup.ResourceNotFoundException:
+				// Non-existent is OK and we skip over them
+				continue
+			default:
+				return nil, err
+			}
+		}
+
+		// Only delete policies that exist.
+		if resp.Policy != nil {
+			resources = append(resources, &BackupVaultAccessPolicy{
+				svc:             svc,
+				backupVaultName: *out.BackupVaultName,
+			})
+		}
+	}
+
+	return resources, nil
+}
+
+func (b *BackupVaultAccessPolicy) Remove() error {
+	// Set the policy to a policy that allows deletion before removal.
+	//
+	// This is required to delete the policy for the automagically created vaults
+	// such as "aws/efs/automatic-backup-vault" from EFS automatic backups
+	// which by default Deny policy deletion via backup::DeleteBackupVaultAccessPolicy
+	//
+	// Example "aws/efs/automatic-backup-vault" default policy:
+	//
+	// {
+	//     "Version": "2012-10-17",
+	//     "Statement": [
+	//         {
+	//             "Effect": "Deny",
+	//             "Principal": {
+	//                 "AWS": "*"
+	//             },
+	//             "Action": [
+	//                 "backup:DeleteBackupVault",
+	//                 "backup:DeleteBackupVaultAccessPolicy",
+	//                 "backup:DeleteRecoveryPoint",
+	//                 "backup:StartCopyJob",
+	//                 "backup:StartRestoreJob",
+	//                 "backup:UpdateRecoveryPointLifecycle"
+	//             ],
+	//             "Resource": "*"
+	//         }
+	//     ]
+	// }
+	//
+	// While deletion is Denied, you can update the policy with one that
+	// doesn't deny and then delete at will.
+	allowDeletionPolicy := `{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "*"
+            },
+            "Action": "backup:DeleteBackupVaultAccessPolicy",
+            "Resource": "*"
+        }
+    ]
+}`
+	_, err := b.svc.PutBackupVaultAccessPolicy(&backup.PutBackupVaultAccessPolicyInput{
+		BackupVaultName: &b.backupVaultName,
+		Policy:          &allowDeletionPolicy,
+	})
+	if err != nil {
```

Comment:
Should this be ignoring the `err` from `PutBackupVaultAccessPolicy`? Someone could allow deletion but not updating the vault access policy. This would fail there.

Reply:
I'm just going to ignore it. Weird config, low chance of this working if PUT isn't allowed but DELETE is, but so be it.